### ISSUE TITLE
Regression after commit f52727c9f

### ISSUE THREAD

**Issue by AlexDaniel** [niner/Inline-Perl5#107]
Problem bisected to f52727c9fc6d2920c6b3c7035bd8bc6682b7da8f.

Here's the code to reproduce the issue:

```perl6
#!/usr/bin/env perl6
use Image::Magick:from<Perl5>;

my $image = Image::Magick.new(size=>‘30x30’);
$image.Read('xc:black');


$image.Draw(fill=>'red',stroke=>'red',primitive=>'line',interpolate=>‘integer’,
            points=>“10,15 9,9 line 20,20 25,30”, strokewidth=>1, antialias=>False);


$image.Draw(fill=>'gold',stroke=>'gold',primitive=>'line',interpolate=>‘integer’,
            points=>“15,18 3,5 line 10,20 25,15”, strokewidth=>1, antialias=>False);

$image.Write('x.png');
```

Note that there are two calls to `.Draw ` so we should get two sets of lines (gold *and* red).

On good revisions (like `f52727c9f^`) I get this image:

![x](https://user-images.githubusercontent.com/5507503/36948209-f0051c28-1fdf-11e8-9595-71d221a4e88a.png)

On bad revisions (starting with f52727c9fc6d2920c6b3c7035bd8bc6682b7da8f) I get this incorrect result:
![x](https://user-images.githubusercontent.com/5507503/36948200-bdb1430a-1fdf-11e8-9208-365137ce4843.png)

**Comment by zoffixznet** [comment]
FWIW, I thought about fixing this by using a `sub` instead of `method`, but in my tests, for some reason not all candidates get added. No idea why:

```
20<Zoffix_>30 m: role Foo {  BEGIN ::?ROLE.^add_multi_method: 'foo', sub (\SELF) { say "no args" }; BEGIN ::?ROLE.^add_multi_method: 'foo', sub (\SELF, |args) { dd args };  BEGIN ::?ROLE.^add_multi_method: 'foo', sub (\SELF, \arg) { say "one arg {arg}" };   }; class :: does Foo  {}.new.foo: :foo
18<camelia> rakudo-moar e9cff7954: OUTPUT: «Unexpected named argument 'foo' passed␤  in sub  at <tmp> line 1␤  in block <unit> at <tmp> line 1␤␤»
```

**Comment by AlexDaniel** [comment]
Can we please revert it first? I'm getting tired of using a local version that is not broken…

**Comment by niner** [comment]
Will see what I can do on the weekend.
Note that a workaround is to quote the argument names: Image::Magick.new('size' => '30x30'); or to not create pairs at all: Image::Magick.new('size', '30x30');

**Comment by niner** [comment]
Should be fixed now in commit 0c9841cc1d5578fd339f4e4e289e87eb780e1dd2
We now check %_ in the generated methods and if named args were passed, we fall back to the generic method call code. Costs some performance, but at least not all of the gains. And there's still a chance that spesh may bring back the performance.
